packages/optimizers/htmlnano/src/HTMLNanoOptimizer.js

```diff
@@ -0,0 +1,38 @@
+// @flow strict-local
+
+// $FlowFixMe this is untyped
+import htmlnano from 'htmlnano';
+import nullthrows from 'nullthrows';
+import {loadConfig} from '@parcel/utils';
+import {Optimizer} from '@parcel/plugin';
+import posthtml from 'posthtml';
+
+export default new Optimizer({
+  async optimize({contents, map, bundle, options}) {
+    if (!options.minify) {
+      return {contents, map};
+    }
+
+    if (typeof contents !== 'string') {
+      throw new Error(
+        'HTMLNanoOptimizer: Only string contents are currently supported'
+      );
+    }
+
+    let userConfig = await loadConfig(
+      options.inputFS,
+      nullthrows(bundle.getMainEntry()).filePath,
```

Comment:
Not all bundles have a main entry, please use `path.join(options.projectRoot, 'index')` for now.